### DISCUSSION TITLE
lots of suggestions - filters, brush things, curves, selection

### DISCUSSION THREAD

**Discussion by goldstargloww** [keiwando/toonsquid#186] [Ideas]
hi! so first of all i?? love this app so far?? but a few things i've noticed so far in my first 3 days of using it
- as far as i can tell there's no way to do gradients (without using soft brushes) nor are there basic filters like gaussian blur for example, i know at least the filter part has been suggested before but i haven't seen much discussion on it
- i've started using the vector brushes and i've noticed that at least with them, they scale with your zoom level—this isn't inherently bad but an option to toggle this would be nice!
- stabilization would be helpful
- this one's more far fetched but exporting/importing brushes to/from different programs like procreate or clip studio paint would be really nice! this would likely be hard but it'd be really useful :>
- being able to make custom easing curves would be really really helpful! alight motion does it pretty well if you're looking for reference
- on the topic of custom curves, custom pen pressure curves would be nice as well
- presets in size and opacity sliders would be helpful, this is something procreate has and it's really helpful :>
- a selection brush would be really nice, where you can draw in what areas you want to select—an eraser to go along with this too would be ideal
- being able to select multiple keyframes or animation layers to copy/paste or move would be really helpful! right now for keyframes i have to copy each keyframe, count the amount of frames it is relative to others, and then paste it, and repeat that for every keyframe i want to copy
- on extending frames, being able to do it before and after instead of just after would be helpful - something like doing extend backward to playhead, move backward to playhead, cut off behind playhead, and the opposite but for forward and in front of would be helpful - fairly low priority tho

**Comment by Lioncub901** [comment]
- There is a gradient under the path tool but not gaussian. Link: https://toonsquid.com/handbook/colors/gradients/
- The point of a vector brush is to be clear even when zoom if you don't like it use raster brushes. Link: https://toonsquid.com/handbook/brushes/vector/
- there is stabilization. In the tool editor there is a slider called smooth. If you turn it up it will stabilize your brush more. Link:  https://toonsquid.com/handbook/brushes/brush_settings/#smoothing
- Yeah I want that too. We need to export.
- Yeah
- Yeah
- I don't know what you mean by selection brush but there is a selection tool and you can remove from a selection. Link: https://toonsquid.com/handbook/selection/selection_tool/
- You can select multiple frame by double tapping fast and then dragging to the frame you want. Link: https://toonsquid.com/handbook/interface/timeline/#multiselection-drawings
- Yeah, but you can just select all the frame behind and select the frame you want to extend and then extend it forward.
-

**Comment by Cat-Butler** [comment]
### Gradients:
The gradients for Path objects (hand-drawn curves, various shapes) work for both Strokes and Fills.
Note that they can also be combined with various masks, blending modes, opacity, and a vector layer with gradients can also be rasterized, converting it to a pixel layer but preserving the gradient.

![9ABDCBD9-F36A-4601-855D-A62DB33DBBC5](https://user-images.githubusercontent.com/25494772/226171141-62d2b667-bb35-4e5d-bee2-90558ce1e4cb.jpeg)

![F3E5F9F8-0E31-4823-8681-3738559F6531](https://user-images.githubusercontent.com/25494772/226171152-e732788d-ba30-4e0e-b2f8-5ad9e15353f7.jpeg)

### Brushes:
.abr brushes can be imported (Adobe Photoshop brush format, there are more out there than I can count - I'll be ignoring comments on my inadequate counting ability), but because they are different brush engines, they might not work how you expect (if at all).
Some brush vendors will provide brushes in Procreate, Photoshop, and/or Affinity format - either as separate products or bundled together (admittedly, some of these are stamp-based).

![24A8B28D-B80B-48DA-AC69-3B4A5BE92A4C](https://user-images.githubusercontent.com/25494772/226171472-d699efc7-6bbe-45ae-b23d-d921d983b410.jpeg)

![5B1C9508-97A6-4F2B-8966-63182424BB28](https://user-images.githubusercontent.com/25494772/226171495-e5922b7e-3a14-4917-94e6-ca2f24290bf3.jpeg)

### Brush presets:
Workaround: brushes will remember size and opacity on a per-brush basis, so you can duplicate brushes to have different sizes (until fairly recently, when presets were introduced, you had to do this in Procreate too), although it's definitely not as convenient as presets within a single brush.

### Multiple keyframes:
I think this is more in regard to selecting multiples of those little diamond keyframes, not drawing frames, and yes, I want this too.
